vscode/microsoft-kiota/src/enums.ts

```diff
@@ -0,0 +1,96 @@
+import { QuickPickItem } from "vscode";
+
+export enum GenerationType {
+    // eslint-disable-next-line @typescript-eslint/naming-convention
+    Client = 0,
+    // eslint-disable-next-line @typescript-eslint/naming-convention
+    Plugin = 1,
+    // eslint-disable-next-line @typescript-eslint/naming-convention
+    ApiManifest = 2,
+};
+
+export enum KiotaGenerationLanguage {
+    // eslint-disable-next-line @typescript-eslint/naming-convention
+    CSharp = 0,
+    // eslint-disable-next-line @typescript-eslint/naming-convention
+    Java = 1,
+    // eslint-disable-next-line @typescript-eslint/naming-convention
+    TypeScript = 2,
+    // eslint-disable-next-line @typescript-eslint/naming-convention
+    PHP = 3,
+    // eslint-disable-next-line @typescript-eslint/naming-convention
+    Python = 4,
+    // eslint-disable-next-line @typescript-eslint/naming-convention
+    Go = 5,
+    // eslint-disable-next-line @typescript-eslint/naming-convention
+    Swift = 6,
+    // eslint-disable-next-line @typescript-eslint/naming-convention
+    Ruby = 7,
+    // eslint-disable-next-line @typescript-eslint/naming-convention
+    CLI = 8,
+};
+
+export enum KiotaPluginType {
+    // eslint-disable-next-line @typescript-eslint/naming-convention
+    OpenAI = 0,
+    // eslint-disable-next-line @typescript-eslint/naming-convention
+    ApiManifest = 1,
+    // eslint-disable-next-line @typescript-eslint/naming-convention
+    ApiPlugin = 2,
+};  
+
+export function parseGenerationType(generationType: string | QuickPickItem | undefined): GenerationType {
```

Comment:
nit: inclusion of the functions inside a file named enums is confusing. We could keep the file name and leave the functions out or rename the file so that it's clear what we should expect to find inside

Reply:
Thought to reduce code movement, having enums and the functions that operate on the enums in the same file would make things lighter. But I guess moving the functions to util.ts would serve long term. Will do this.